jubilant/_juju.py

```diff
@@ -837,6 +857,142 @@ def scp(
 
         self.cli(*args)
 
+    def secrets(self) -> list[Secret[Hidden]]:
+        """Get all secrets in the model.
+
+        Returns:
+            A list of :class:`Secret[Hidden]` objects, one for each secret in the model.
+        """
+        stdout = self.cli('secrets', '--format', 'json')
+        output: dict[str, _SecretResponse] = json.loads(stdout)
+        return [
+            Secret[Hidden](
+                uri=SecretURI('secret:' + uri_from_juju),
+                name=obj.get('name'),
+                label=obj.get('label'),
+                owner=obj.get('owner'),
+                rotates=datetime.datetime.fromisoformat(obj['rotates'].replace('Z', '+00:00'))
```

Comment:
I'd prefer this wrapped in a little `_datetime_from_iso` helper which took `str | None` and returned `datetime | None`. Then you could use it in all these places, and here with:

```suggestion
                rotates=_datetime_from_iso(obj.get('rotates')),
```